TBHelper conversion

aehart opened the pull request:
Resolves #176 for the TrackBuilder. Other modules can push to this PR.

bryates commented on TrackletAlgorithm/MatchCalculator.h:
```diff
@@ -816,49 +816,49 @@ void MatchCalculator(BXType bx,
         case 3:
         if(FMMask<LAYER, PHISEC, TF::L5L6>()) {
-          fullmatch[3].write_mem(bx,bestmatch,nmcout4); // L5L6 seed
+          fullmatch[FMCount<LAYER, PHISEC, TF::L5L6>()].write_mem(bx,bestmatch,nmcout4); // L5L6 seed
```
This is a some what complicated solution I came up with to count how many slots are not masked. That way, seeding in L5L6 (which would be slot 3) is placed in the correct slot (slot 2 if the TBHelper is only reading L1L2 and L5L6). This relies on the mask to agree with the number of files TBHelper loads, which seems reasonable.

aehart replied:
@bryates I'm not sure I understand what's happening here. I thought the `FMMask` essentially told you which FM memories are "on" for a given MC. That way, the size of the `fullmatch` array can be the same for every MC, but HLS will only generate ports for the memories that are enabled by the mask.

So why would the index of, e.g., the L5L6 FM memory ever change?

bryates replied:
`FMMask` disables ports so Vivado will trim them, but this does not change the index, which was hard coded to 3 in the past. `FMCount` allows it to determine how many ports are actually active, and allocate the correct spot. I needed this because the `TBHelper` uses vectors that are only as large as the active ports.

aehart commented:
FYI, I just rebased and removed the commits related to the MatchProcessor, so as to expedite the review for modules in the summer chain.

@bryates I moved the MP commits to the mp_tb_helper branch. Please wait until this PR is merged to open a PR for that branch.
